regressiontests/regression_test.go

```diff
@@ -0,0 +1,348 @@
+				continue
+			} else {
+				t.Logf("Synced to tip of chain: %d", localTipHeight)
+				break out
+			}
+		}
+	}
+
+	t.Log("Comparing synced data with known checkpoints...")
+
+	verifyHeaders(t)
+
+	t.Log("Headers checkpoint comparison passed successfully 🎉")
+
+	t.Log("Verifying merkle roots...")
+
+	verifyMerkleRoots(t, merkleRootFixtures)
+
+	t.Log("Merkle roots verification passed successfully 🎉")
+}
+
+func startPostgresContainer(ctx context.Context, t *testing.T) (testcontainers.Container, string) {
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:latest",
+		ExposedPorts: []string{"5432/tcp"},
+		Env: map[string]string{
+			"POSTGRES_USER":     "testuser",
+			"POSTGRES_PASSWORD": "test",
+			"POSTGRES_DB":       "bhs",
+		},
+		WaitingFor: wait.ForListeningPort("5432/tcp"),
+	}
+	postgresContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		t.Fatalf("Could not start postgres container: %v", err)
+	}
+
+	time.Sleep(10 * time.Second) // Wait for the container to be ready
+
+	mappedPort, err := postgresContainer.MappedPort(ctx, "5432")
+	if err != nil {
+		t.Fatalf("Failed to get mapped port: %v", err)
+	}
+	return postgresContainer, mappedPort.Port()
+}
+
+func fetchExternalForkHeight(url string) (int, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return 0, fmt.Errorf("failed to make HTTP request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var forks []WhatsOnChainForkTip
+	if err := json.NewDecoder(resp.Body).Decode(&forks); err != nil {
+		return 0, fmt.Errorf("failed to decode JSON response: %w", err)
+	}
+
+	for _, fork := range forks {
+		if fork.Status == "active" {
+			return fork.Height, nil
+		}
+	}
+
+	return 0, fmt.Errorf("no active fork found")
+}
+
+func fetchLocalTip() (int, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://localhost:8080/api/v1/chain/tip/longest", nil)
+	if err != nil {
+		return 0, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Authorization", "Bearer "+config.DefaultAppToken)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return 0, fmt.Errorf("failed to make HTTP request to application: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		body, _ := io.ReadAll(io.Reader(resp.Body))
+		return 0, fmt.Errorf("unexpected status code: %d, body: %s", resp.StatusCode, string(body))
+	}
+
+	var chainTip tips.TipStateResponse
+	if err := json.NewDecoder(resp.Body).Decode(&chainTip); err != nil {
+		return 0, fmt.Errorf("failed to decode JSON response: %w", err)
+	}
+
+	return int(chainTip.Height), nil
+}
+
+func fetchBlockHeader(hash string) (*headers.BlockHeaderStateResponse, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://localhost:8080/api/v1/chain/header/state/"+hash, nil)
```

Comment:
Consider make a global constant: `http://localhost:8080` and re use it everywhere it's applicable.

Reply:
done